presto-verifier/src/main/java/com/facebook/presto/verifier/Validator.java

```diff
@@ -182,7 +182,60 @@ public boolean isDeterministic()
 
     private boolean validate()
     {
-        controlResult = executeQuery(controlGateway, controlUsername, controlPassword, queryPair.getControl(), controlTimeout, sessionProperties);
+        boolean result = false;
```

Comment:
This needs to be restructured a little. One of the things that the verifier does is re-run the control query a couple times to see if the results are non-deterministic, before declaring that the results are not correct. However, that needs to re-run the setup query as well, otherwise it won't redo the INSERT, it will just re-run the SELECT step. See below for where I would move this to